modules/op-using-secrets-and-workspaces.adoc

```diff
@@ -0,0 +1,367 @@
+
+To ensure that the task takes this optional property into account the `git-clone` task is adapted to include an optional workspace. The `apiversion`, `kind`, `metadata`, `params`, `results`, and `steps` of the `git-clone` task are defined as in the standard task, but two workspaces are defined, one of which is optional. The `name` and `description` of the optional workspace are defined and the script is updated to conditionally copy the content of the workspace to the user’s home directory only if the workspace is bound.
+
+The advantage of using optional workspaces is that it makes the task more flexible for example if a public Git repository is cloned, there s no need to bind a secret to a workspace. 
+
+The following steps illustrate how to use the optional workspace feature in the git-clone Task:
+
+. Create a Task called `git-clone-optional-ws` that clones a git repository using SSH authentication, with an optional workspace. 
+
+. Define workspaces for the `git-ssh` secret and the optional workspace `my-optional-workspace`. 
+
+. Define task parameters for the repository URL, revision, and the git-init image that the Task uses. 
+
+. Define results that the Task produces, for example, the commit SHA and the repository URL. 
+
+. Define the steps that the Task executes, including a condition to check if the optional workspace is bound before copying the content of the workspace to the user's home directory. 
+
++
+[source,yaml,subs="attributes+"]
+----
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: git-clone-optional-ws
+spec:
+  workspaces:
+    - name: git-ssh
+      description: Workspace containing the git-ssh secret
+    - name: my-optional-workspace
+      description: Optional workspace that may be bound to a Pod or Container
+      optional: true
+  params:
+    - name: url
+      description: The URL of the Git repository to clone
+      type: string
+      default: ""
+    - name: revision
+      description: The Git revision to check out
+      type: string
+      default: "master"
+    - name: image
+      description: The image to use for the git-init container
+      type: string
+      default: "alpine/git:latest"
+  results:
+    - name: commit-sha
+      description: The commit SHA of the checked out revision
+    - name: repo-url
+      description: The URL of the cloned Git repository
+  steps:
+    - name: clone
+      image: $(params.image)
+      workingDir: /workspace/source
+      env:
+        - name: GIT_SSH_COMMAND
+          value: ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i /workspace/git-ssh/ssh-privatekey
+      command:
+        - sh
+      args:
+        - -c
+        - |
+          git clone $(params.url) .
+          git checkout $(params.revision)
+          echo $(git rev-parse HEAD) > $(results.commit-sha.path)
+          echo $(params.url) > $(results.repo-url.path)
+    - name: optional-workspace
```

Comment:
This step doesn't make sense - it looks like it exists just to demonstrate this feature but it doesn't do anything useful.

IMHO, we could again use the `git-clone` task that we ship with product as it demonstrates this feature in useful way.